### PR TITLE
TP-1134 Add OTHER logic to radio buttons

### PR DESCRIPTION
**Pull request by aktfrikshun**
TP-1134 Add OTHER logic to radio buttons

### CODE REVIEW

File: app/views/components/forms/question_types/_radio_buttons.html.erb [hunk 1]
```diff
@@ -9,6 +9,10 @@
       <%= label_tag(nil, for: "#{question.answer_field}_#{(index + 1).to_s}", class: "usa-radio__label") do %>
         <%= option.text %>
       <% end %>
+      <% if ["OTHER", "OTRO"].include?(option.text.upcase) %>
```
Comment (ryanwoldatwork) [inline app/views/components/forms/question_types/_radio_buttons.html.erb:12]: this works if the 'other' radio is selected, then the value will be 'other', even if there is text in the text field.

the value of the response should be the typed in value, or if blank, then the value of the `other` text

Comment (ryanwoldatwork) [reply app/views/components/forms/question_types/_radio_buttons.html.erb:12]: and, got specs? 

🔴  🟢

Comment (aktfrikshun) [reply app/views/components/forms/question_types/_radio_buttons.html.erb:13]: Confirmed radio button works with specs

File: app/views/components/forms/question_types/_checkbox.html.erb [hunk 2]
```diff
@@ -11,7 +11,7 @@
         <%= option.text %>
       <% end %>
       <% if ["OTHER", "OTRO"].include?(option.text.upcase) %>
-        <input type="text" name="<%= question.answer_field %>_other" id="<%= question.answer_field %>_other" data-checkbox-id="<%= @option_id %>" placeholder="<%= t 'form.enter_other_text' %>" class="usa-input other-option" />
```
Comment (ryanwoldatwork) [inline app/views/components/forms/question_types/_checkbox.html.erb:14]: 👍

File: spec/factories/form.rb [hunk 3]
```diff
@@ -180,5 +180,19 @@
...
     end
 
+    trait :radio_button_form do
+      name { "Radio button form" }
+      kind { "custom" }
+      after(:create) do |f, evaluator|
+        FactoryBot.create(:question,
+          :with_radio_buttons,
+          form: f,
+          answer_field: :answer_03,
```
Comment (ryanwoldatwork) [inline spec/factories/form.rb:190]: Why `_03` instead of `_01`?

**Pull request merged**